playbooks/roles/backups/templates/backup.py

```diff
@@ -0,0 +1,324 @@
+        backup_path = os.path.join(backup_dir, backup_filename)
+        cmd = cmd_template.format(backup_path)
+
+        logging.info('Dumping database: `{}`'.format(cmd))
+        ret = subprocess.call(cmd, shell=True)
+        if ret:  # if non-zero return
+            error_msg = 'Error occurred while dumping database'
+            logging.error(error_msg)
+            raise Exception(error_msg)
+
+        return backup_path
+    else:
+        error_msg = 'Unknown service {}'.format(service_name)
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+
+def clean_up(backup_path):
+    """
+    Remove the local database dump and the corresponding tar file if it exists.
+
+        backup_path: An absolute path to a file or directory containing a
+            database dump.
+    """
+
+    logging.info('Cleaning up "{}"'.format(backup_path))
+
+    backup_tar = backup_path + '.tar.gz'
+    if os.path.isfile(backup_tar):
+        os.remove(backup_tar)
+
+    if os.path.isdir(backup_path):
+        shutil.rmtree(backup_path)
+    elif os.path.isfile(backup_path):
+        os.remove(backup_path)
+
+
+def restore(service_name, backup_path, uncompress=True):
+    """
+    Restore a database from a backup.
+
+        service_name: The name of the service whose database is to be restored,
+            either mysql or mongodb.
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    if service_name == 'mongodb':
+        restore_mongodb(backup_path, uncompress)
+    elif service_name == 'mysql':
+        restore_mysql(backup_path, uncompress)
+
+
+def restore_mongodb(backup_path, uncompress=True):
+    """
+    Restore a MongoDB database from a backup.
+
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    logging.info('Restoring MongoDB from "{}"'.format(backup_path))
+
+    if uncompress:
+        backup_path = _uncompress(backup_path)
+
+    cmd = 'mongorestore {}'.format(backup_path)
+    ret = subprocess.call(cmd, shell=True)
+    if ret:  # if non-zero return
+        error_msg = 'Error occurred while restoring MongoDB backup'
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    logging.info('MongoDB successfully restored')
+
+
+def restore_mysql(backup_path, uncompress=True):
+    """
+    Restore a MySQL database from a backup.
+
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    logging.info('Restoring MySQL from "{}"'.format(backup_path))
+
+    if uncompress:
+        backup_path = _uncompress(backup_path)
+
+    cmd = 'mysqladmin -f drop edxapp'
```

Comment:
+1 for Google cloud sql. As for dropping....You have a point that in a disaster scenario we would want to drop everything. But perhaps there are scenarios that we can't think of right now where a restore would fail (or be inadequate for whatever reason) and that we would have to restore the previous state.